pkg/parser/terraform/terraform.go

```diff
@@ -36,6 +37,36 @@ func (p *Parser) Resolve(fileContent []byte, filename string) (*[]byte, error) {
 	return &fileContent, nil
 }
 
+func addExtraInfo(json []model.Document, path string) []model.Document {
+	var resourcesElements model.Document
+	var elements model.Document
+	var certInfo map[string]interface{}
+
+	for _, documents := range json { // iterate over documents
+		if documents["resource"] != nil {
+			for _, resources := range documents["resource"].(model.Document) { // iterate over resources
+				resourcesElements = resources.(model.Document)
+				for _, v2 := range resourcesElements { // resource name
+					elements = v2.(model.Document)
+					for k, v3 := range elements { // resource elements
+						if k == "certificate_body" {
+							content := additional.CheckCertificate(v3.(string))
+							if content != "" {
+								certInfo = additional.AddCertificateInfo(path, content)
+								if certInfo != nil {
+									elements["certificate_body"] = certInfo
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
```

Comment:
Extract nested for loops to separated functions passing down parameters and returning values as needed. I could only find an example of this in Python https://codereview.stackexchange.com/a/191823